File: Problems/024-swap-nodes-in-pairs/swap-nodes-in-pairs.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        dummy = ListNode(0)
        dummy.next = head
        p = dummy
        
        while p.next and p.next.next:
            a, b = p.next, p.next.next
            a.next = b.next
            b.next = a
            p.next = b
            p = p.next.next

        return dummy.next

File: Problems/024-swap-nodes-in-pairs/swap-nodes-in-pairs.go
package problem024

// ListNode definition.
type ListNode struct {
	Val  int
	Next *ListNode
}

func swapPairs(head *ListNode) *ListNode {
	dummy := &ListNode{}
	dummy.Next = head
	p := dummy

	for p.Next != nil && p.Next.Next != nil {
		a, b := p.Next, p.Next.Next
		a.Next = b.Next
		p.Next = b
		b.Next = a
		p = p.Next.Next
	}

	return dummy.Next
}
